container/openejb-core/src/main/java/org/apache/openejb/core/ivm/naming/IvmContext.java

```diff
@@ -477,9 +489,36 @@ public String getNameInNamespace() throws NamingException {
     public void close() throws NamingException {
     }
 
-    protected void checkReadOnly() throws OperationNotSupportedException {
+    //TODO: rename? isReadOnly?
+    /*
+     * return false if current naming context is not marked as read only
+     * return true if current naming context is marked as read only and system property jndiExceptionOnFailedWrite is set to false
+     * 
+     * throws OperationNotSupportedException if naming context:
+     *   - is marked as read only and
+     *   - system property jndiExceptionOnFailedWrite is set to true
+     *   
+     * jndiExceptionOnFailedWrite property is defined by tomcat and is used in similar context for web app components
+     * https://tomcat.apache.org/tomcat-7.0-doc/config/context.html#jndiExceptionOnFailedWrite
+     * 
+     */
+    protected boolean checkReadOnly() throws OperationNotSupportedException {
+      //TODO: should it log?
         if (readOnly) {
-            throw new OperationNotSupportedException();
+            //alignment with tomcat behavior
+            if("true".equals(System.getProperty("jndiExceptionOnFailedWrite"))) {
```

Comment:
SystemInstance.get().getProperty and openejb prefix as well, default should fail as before